fdbserver/worker.actor.cpp

```diff
@@ -1181,7 +1211,7 @@ ACTOR Future<Void> workerServer(
 			when(state ExecuteRequest req = waitNext(interf.execReq.getFuture())) {
 				state ExecCmdValueString execArg(req.execPayload);
 				try {
-					int err = wait(execHelper(&execArg, coordFolder, "role=coordinator"));
+					int err = wait(execHelper(&execArg, coordFolder, "role=coordinator", 1 /*version*/));
```

Comment:
This is a really bad idea - this will cause the actor to block until the execution finished. During this time it won't react to any changes to `serverDBInfo` which will cause longer recovery-times if something fails during a snapshot.

Instead you should put this into an `ActorCollection` and wait on that (and `ASSERT(false)` if it ever finishes)

The general rule is: if you have a `chose-when` construct you almost never want to call `wait` within a `when` block

Reply:
Yes, this is Snap V1 code which is going to be removed once V2 is official. I was planning on either fixing if Snap V2 does not materialize. I will add a FIXME.